azure-iot-device/azure/iot/device/common/pipeline/pipeline_stages_base.py

```diff
@@ -1009,15 +1008,21 @@ def _handle_pipeline_event(self, event):
                 )
             )
 
-            if self.pipeline_root.connected and self.state == ReconnectState.LOGICALLY_CONNECTED:
+            # QUESTION: why do we check the pipeline root connected here? Is this necessary or just adding extra edge cases?
+            # In what situation would the pipeline root ever be disconnected while the pipeline state is logically connected?
```

Comment:
># In what situation would the pipeline root ever be disconnected while the pipeline state is logically connected? [](http://example.com/codeflow?start=12&length=113)

If the connection is already dropped, root.connected will be False and state will be LOGICALLY_CONNECTED.  It is possible to get an on_disconnect callback when the transport is already disconnected.